web/app/page.js

```diff
@@ -58,5 +64,40 @@ export default function Home() {
 }
 
 function allergenTimeSortCallback(a, b) {
-  return b.time - a.time;
+  if (a.time && b.time) {
+    return a.time - b.time;
+  }
+  if (!a.time) {
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
+function loadJSONfromLocal(key) {
+  const storedTimes = JSON.parse(window.localStorage.getItem(key));
+  if (storedTimes === null) {
+    console.log("LoadJSON reading null");
+    return null;
+  }
+  return storedTimes.map(({ name, time }) => {
+    if (time) {
+      return { name, time: new Date(time) };
+    }
+    return { name, time };
```

Comment:
```suggestion
      return { name, time: (time ? new Date(time) : time)};
```

Reply:
We can't omit time because we need to change it from a string to a date object here. I can refactor it though.